src/validators/union.rs

```diff
@@ -416,14 +464,20 @@ impl TaggedUnionValidator {
         if let Some(validator) = self.choices.get(tag.as_ref()) {
             return match validator.validate(py, input, extra, slots, recursion_guard) {
                 Ok(res) => Ok(res),
-                Err(err) => Err(err.with_outer_location(tag.as_ref().into())),
+                Err(err) => match tag.as_ref() {
+                    ChoiceKey::Str(s) => Err(err.with_outer_location((*s).clone().into())),
+                    ChoiceKey::Int(i) => Err(err.with_outer_location((*i).clone().to_string().into())),
+                },
             };
         } else if let Some(ref repeat_choices) = self.repeat_choices {
             if let Some(choice_tag) = repeat_choices.get(tag.as_ref()) {
                 let validator = &self.choices[choice_tag];
                 return match validator.validate(py, input, extra, slots, recursion_guard) {
                     Ok(res) => Ok(res),
-                    Err(err) => Err(err.with_outer_location(tag.as_ref().into())),
+                    Err(err) => match tag.as_ref() {
+                        ChoiceKey::Str(s) => Err(err.with_outer_location((*s).clone().into())),
```

Comment:
```suggestion
                        ChoiceKey::Str(s) => Err(err.with_outer_location(s.into())),
```
Should work since we have `impl From<&str> for LocItem`, and above.

Reply:
actually `s` is a `&string` not a `&str` so it doesn't compile sadly

Reply:
I've fixed this.